utils/create-component.js

```diff
@@ -0,0 +1,56 @@
+require("colors");
+const fs = require("fs");
+const templates = require("./templates");
+
+const componentName = process.argv[2];
+const componentPath = process.argv[3];
+
+if (!componentName) {
+  console.error("Please supply a valid component name".red);
+  process.exit(1);
+}
+
+console.log("Creating Component Templates with name: " + componentName);
+
+let componentDirectory
+if (!componentPath) {
+  componentDirectory = `./src/components/${componentName}`;
+} else {
+  if (!fs.existsSync(`./src/${componentPath}`)) {
+    fs.mkdirSync(`./src/${componentPath}`)
+  }
+
+  componentDirectory = `./src/${componentPath}/${componentName}`;
+}
+
+if (fs.existsSync(componentDirectory)) {
+  console.error(`Component ${componentName} already exists.`.red);
+  process.exit(1);
+}
+
+fs.mkdirSync(componentDirectory);
+
+const generatedTemplates = templates.map((template) => template(componentName));
+
+generatedTemplates.forEach((template) => {
+  if (template.type === 'style') {
+    fs.writeFileSync(
+      `${componentDirectory}/${componentName}${template.extension}`,
+      template.content
+    )
+  } else if (template.type === 'index') {
+    fs.writeFileSync(
+      `${componentDirectory}/index${template.extension}`,
+      template.content
+    );
+  } else {
+    fs.writeFileSync(
+      `${componentDirectory}/${componentName}${template.extension}`,
+      template.content
+    );
+  }
```

Comment:
If there is an existing file at that path, will it overwrite it?

Reply:
If the specified path exists, it creates a component within that path.

`npm run generate Test atoms/Test`

it will generate in `src/components/atoms/Test/Test`

if the `.../Test/Test` path already exists, it generates one level down so: `src/components/atoms/Test/Test/Test`

That case is, i believe, a user error for which there shouldnt be too much time spent fixing.
The fix for this is to just amend the `/Test` from the path argument